src/utils/propNormalizer.ts

```diff
@@ -0,0 +1,47 @@
+import { camelize } from '@vue/shared';
+
+/**
+ * Checks for duplicate prop definitions and logs a warning if found
+ * @param camelKey - The camelCase version of the prop name
+ * @param currentKey - The current prop key being processed
+ * @param keyMapping - Mapping of camelCase keys to their original keys
+ */
+function checkForDuplicateProp(
+  camelKey: string,
+  currentKey: string,
+  keyMapping: Record<string, string>
+): void {
+  if (camelKey in keyMapping && currentKey !== keyMapping[camelKey]) {
+    console.warn(
```

Comment:
What does Vue do for 'normal' components, if there are duplicate keys? Log a warning, throw an error, or nothing? I think we should do the same.